server/api/src/routes/expenses.rs

```diff
@@ -1,23 +1,39 @@
 use sea_orm::DatabaseConnection;
-use veryrezsi_core::dto::expenses::{NewExpenseRequest, NewPredefinedExpenseRequest};
-use veryrezsi_core::logic::{expense_operations, user_operations};
+use veryrezsi_core::dto::expenses::{
+    ExpenseResponse, ExpenseWithTransactions, NewExpenseRequest, NewPredefinedExpenseRequest,
+};
+use veryrezsi_core::logic::{expense_operations, transaction_operations};
 
 use super::common::ValidatedJson;
 use super::error::ErrorMsg;
 use crate::auth;
 
 use axum::extract::{Path, State};
 use axum::Json;
-use entity::{expense, predefined_expense, Id};
+use entity::{predefined_expense, Id};
 
-// FIXME: Not a good idea to return with domain model
 pub async fn get_expenses(
     user: auth::AuthenticatedUser,
     State(ref conn): State<DatabaseConnection>,
     Path(user_id): Path<Id>,
-) -> Result<Json<Vec<expense::Model>>, ErrorMsg<()>> {
+) -> Result<Json<ExpenseResponse>, ErrorMsg<()>> {
     match expense_operations::find_expenses_by_user_id(conn, user.id, user_id).await {
-        Ok(expenses) => Ok(Json(expenses)),
+        Ok(expenses) => {
+            let mut expense_response: ExpenseResponse = vec![];
+
+            for expense in expenses {
```

Comment:
I am not exactly sure I like DTO construction in the "controller" layer but maaaaybe it is good? I genuinely don't know.

Reply:
Refactored it into logic.